hw/ip/clkmgr/doc/_index.md

```diff
@@ -0,0 +1,191 @@
+| Peripheral      | 24MHz, 48MHz, 96MHz            | I2c, Spi, Uart, Usb, others                                    | Yes            | Yes                |
+| Timers          | 100-200KHz, 24MHz              | AON timers, Timers, Watchdog                                   | No             | No                 |
+
+* 1 - Transactional clock group's software control is only a software hint.
+* 2 - Transactional clock group's wait-for-interrupt control is only a hint.
+* 3 - May require additional complexity to handle multi-host (non-wait-for-interrupt) scenarios
+
+## Power-up Clock Group
+
+The group refers to modules responsible for power up, such as power, reset and clock managers.
+Large portions of these modules operate to release clocks and resets for the rest of the design, thus cannot operate on gated versions of the clocks themselves.
+They are the only group running clocks directly from the source.
+All follow groups are derived after root clock gating.
+See [block diagram]() for more details
+
+## Transactional Clock Group
+
+This group refers to the collection of modules that are transactional by nature (example: `Hmac` / `Aes` / `Kmac`).
+This means these modules perform specific tasks (for example encrypt, decrypt or hashing).
+While performing these tasks, it is unsafe to manipulate or change the clocks.
+Once these tasks are complete, the clocks can be safely shut-off.
+
+To ensure such behavior on the clocks, The final clock enable is qualified with an `Idle` indication to signify that a transaction is ongoing and manipulation of the clock is not permitted.
+The `Idle` signal must be sourced from the transactional modules and sent to the clock manager.
+
+For this group software can only express its intent to shut-off, and does not have full control over the final state.
+This intent is indicated with a register in the clock manager register file, see {{< regref "CLK_HINTS" >}}.
+
+Wait-for-interrupt based control is already a software hint, it can thus be applied to this group with the same `Idle` requirement.
+
+For modules in this group, each module can be individually influenced, and thus each has its own dedicated clock gating logic.
+The added benefit of allowing software control in this group is to save power, as some transactional modules can be both power and area hungry.
+
+## Infrastructure Clock Group
+
+This group refers to the collection of modules that support infrastructure functions.
+
+If the clocks to these modules are turned off, there may not be a way to turn them back on and could thus result in system deadlock.
+This includes but is not limited to:
+* Turning off fabric / gasket clocks, and thus having no way to access the fabric and resume the clock.
+* Turning off memory clocks such that there is no way to execute code that would resume the clocks.
+
+For this group, there is no reason to allow software control over the clocks, as it could be used to commit suicide.
```

Comment:
Nit: we could do with more accurate phrasing than "commit suicide" - maybe say "as it could be used to put the chip in a deadlock state only recoverable by reset" or something along those lines? Maybe "deadlock" is a reasonable term to include? I'm not sure.

Reply:
yes sounds good.  The wording i chose probably is not appropriate anyways, so I should re-word.